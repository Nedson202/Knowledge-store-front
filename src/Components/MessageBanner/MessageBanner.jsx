import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const MessageBanner = (props) => {
  const { isAuthenticated, user } = props;

  const emailConfirmationNote = () => (
    <div className="pending-verification">
      A verification link has been sent to your mail box.
      Click on the link or use this
      &nbsp;
      <Link
        to="/email?verify-email=true"
        style={{ color: 'black' }}
      >
        Link
      </Link>
      &nbsp;
      to request for a new verification mail.
    </div>
  );

  if (!isAuthenticated || user.isVerified) {
    return <></>;
  }

  return (
    <Fragment>
      {emailConfirmationNote()}
    </Fragment>
  );
};

MessageBanner.propTypes = {
  isAuthenticated: PropTypes.bool,
  user: PropTypes.object,
};

MessageBanner.defaultProps = {
  isAuthenticated: false,
  user: {},
};

const mapStateToProps = ({ auth }) => ({
  isAuthenticated: auth.isAuthenticated,
  user: auth.user,
});

export default connect(mapStateToProps)(MessageBanner);
