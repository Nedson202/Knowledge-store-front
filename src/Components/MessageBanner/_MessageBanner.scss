.pending-verification {
  position: fixed;
  width: 100%;
  top: 0;
	height: 50px;
	text-align: center;
	line-height: 50px;
	font-size: 18px;
	background-color: $yellow_2;
  color: $white_1;
  z-index: 20;
}

.pending-verification +
.navbar,

.pending-verification +
.navbar +
.leftsidebar,

.pending-verification +
.navbar +
.leftsidebar +
.container-content {
  margin-top: 50px;
}

@media screen and (max-width: 560px) {
  .pending-verification {
    height: unset;
    text-align: center;
    line-height: unset;
    font-size: 17px;
    background-color: $yellow_2;
    color: $white_1;
    padding: 10px 15px;
  }
}
