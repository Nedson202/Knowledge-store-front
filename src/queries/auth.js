import { gql } from 'apollo-boost';

const addUser = gql`
  mutation($username: String!, $password: String!, $email: String!) {
    addUser(username: $username, password: $password, email: $email) {
      token
    }
  }
`;

const verifyEmail = gql`
  mutation($token: String!) {
    verifyEmail(token: $token) {
      token,
      message
    }
  }
`;

const sendVerificationEmail = gql`
  query($email: String!) {
    sendVerificationEmail(email: $email) {
      message
    }
  }
`;

const forgotPassword = gql`
  query($email: String!) {
    forgotPassword(email: $email) {
      message
    }
  }
`;

const resetPassword = gql`
  mutation($id: String!, $email: String!, $password: String!, $token: String!) {
    resetPassword(id: $id, email: $email, password: $password, token: $token) {
      message
      token
    }
  }
`;

const editProfile = gql`
  mutation($username: String, $email: String, $picture: String) {
    editProfile(username: $username, email: $email, picture: $picture) {
      token
      message
    }
  }
`;

const changePassword = gql`
  mutation($oldPassword: String!, $newPassword: String!) {
    changePassword(oldPassword: $oldPassword, newPassword: $newPassword) {
      token
      message
    }
  }
`;

const loginUser = gql`
mutation($username: String!, $password: String!) {
  loginUser(username: $username, password: $password) {
    username,
    email,
    token
  }
}
`;

const getBooks = gql`
  {
    books {
      name,
      authors,
      genre,
      year
    }
  }
`;

export {
  addUser, loginUser, getBooks, changePassword,
  verifyEmail, sendVerificationEmail,
  forgotPassword, resetPassword, editProfile
};