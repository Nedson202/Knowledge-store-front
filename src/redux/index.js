import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from './reducers/rootReducer';

const enhancer = composeWithDevTools(applyMiddleware(thunk));

const store = createStore(
  rootReducer,
  enhancer
);

export default store;
