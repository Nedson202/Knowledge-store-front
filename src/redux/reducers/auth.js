import isEmpty from 'lodash/isEmpty';
import { SET_LOGGEDIN_USER } from '../constants/actionTypes';

const initialState = {
  isAuthenticated: false,
  user: {}
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_LOGGEDIN_USER: {
      return {
        ...state,
        isAuthenticated: action.user.id && !isEmpty(action.user),
        user: action.user
      };
    }

    default: return state;
  }
};