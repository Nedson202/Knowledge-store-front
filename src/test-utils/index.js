import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux';

// eslint-disable-next-line react/prop-types
export const AllProviders = ({ children, customStore, apolloMocks }) => (
  <MockedProvider
    addTypename={false}
    mocks={apolloMocks}
  >
    <Provider store={customStore || store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  </MockedProvider>
);

export * from '@testing-library/react';

export const waitForTime = (waitFor = 0) => new Promise(resolve => setTimeout(resolve, waitFor));
