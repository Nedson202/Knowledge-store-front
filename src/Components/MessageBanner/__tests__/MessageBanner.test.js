import React from 'react';
import { createStore } from 'redux';

import { render, AllProviders } from 'test-utils';

import MessageBanner from '../MessageBanner';
import auth from '../../../redux/reducers/auth';

describe('MessageBanner', () => {
  it('should render with default store', () => {
    const { container, asFragment } = render(
      <AllProviders>
        <MessageBanner />
      </AllProviders>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container).toBeInTheDocument();
  });

  it('should display Message Banner if user is not verified', () => {
    const mockProps = {
      auth: {
        isAuthenticated: true,
        user: {
          isVerified: false,
        }
      }
    };

    const store = createStore(auth, mockProps);

    const { getByTestId } = render(
      <AllProviders customStore={store}>
        <MessageBanner />
      </AllProviders>
    );

    getByTestId('banner-message');
  });

  it('should not display Message Banner if user is verified', () => {
    const mockProps = {
      auth: {
        isAuthenticated: true,
        user: {
          isVerified: true,
        }
      }
    };

    const store = createStore(auth, mockProps);

    const { queryByTestId } = render(
      <AllProviders customStore={store}>
        <MessageBanner />
      </AllProviders>
    );

    const banner = queryByTestId('banner-message');

    expect(banner).toBeNull();
  });
});
