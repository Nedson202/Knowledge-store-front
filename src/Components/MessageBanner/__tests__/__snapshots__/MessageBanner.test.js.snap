// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBanner should render with default store 1`] = `<DocumentFragment />`;
